### PR TITLE
Armor upgrade lock element toggle.

### PR DESCRIPTION
**Pull request by ryan-rushton**
Getting this up now as I don't think I will have much DIM time over the next couple of weeks.

Basically I have
- Added a lot element toggle to the armour upgrade drawer.
- Removed the upgrade spend option that had lock element built in.
- Added a new translation for adding + Lock Element to the select option.
- Added a step when loading the spend tier to check if the option still lives. If not we set it to no upgrades.

I am personally fairly impartial to these changes but @nev-r and @inexorableAce seemed pretty confident they were needed. We got requests for this pretty much immediately after the last changes went out.

Screen Shot 2021-07-01 at 7 35 51 pm

Screen Shot 2021-07-01 at 7 35 59 pm

### CODE REVIEW

**Comment by bhollis** [comment]
Should the toggle be positive ("Allow switching element to fit mods")?

**Comment by ryan-rushton** [comment]
I don't have a preference personally. Happy to go with whichever.

**Comment by nev-r** [comment]
- "lock" seems like the clearest brief description to me, versus "allow etc etc etc"
- a sentence-long description accompanying the toggle would be even better.
- i foresee complaints about the material strategy descriptions claiming that they will change elements, even when lock is on. i still believe these long descriptions should be broken up into smaller reusable i18n chunks.

**Comment by ryan-rushton** [comment]
> i foresee complaints about the material strategy descriptions claiming that they will change elements, even when lock is on. i still believe these long descriptions should be broken up into smaller reusable i18n chunks.

I don't believe this is possible at all with the changes.

**Comment by ryan-rushton** [comment]
@bhollis @inexorableAce @nev-r thoughts on this? I think the text changing on screen makes it obvious enough?


https://user-images.githubusercontent.com/7344652/124337855-19ff7600-dbe8-11eb-93c3-a01a230b6e02.mp4

**Comment by inexorableAce** [comment]
Yeah that's great and should help a lot I'd think.

**Comment by nev-r** [comment]
looks great yeah!

File: config/i18n.json [hunk 1]
```diff
@@ -455,15 +454,17 @@
     "CompareLoadout": "Compare Loadout",
     "ConfirmOverwrite": "Are you sure you want to replace the armor in the loadout \"{{name}}\" with this new set of armor?",
     "CreateLoadout": "Save Loadout",
+    "ElementLockedDescription": ". All items will use their existing element.",
+    "ElementNotLockedDescription": ", and will change element as needed to assign mods.",
```
Comment (nev-r) [inline config/i18n.json:458]:
```suggestion
    "ElementNotLockedDescription": "Items will change element as needed to assign mods.",
```
for i18n purposes (clarity for translators, grammar & syntax differences) i would stick with making these both full sentences.

**Review by nev-r — APPROVED** [review]

File: config/i18n.json (hunk 1 above)
Comment (ryan-rushton) [reply config/i18n.json:458]: Yeah I totally agree. I very nearly made two of each overarching translation so the translators had full context in each case. Making them full sentences is a nice halfway.

**Comment by ryan-rushton** [comment]
Going to merge this after release

**Comment by bhollis** [comment]
Please go ahead and merge this.

**Comment by ryan-rushton** [comment]
Sorry, I have been a bit busy and haven't gotten around to it. Sorting it out now.

**Pull request merged**